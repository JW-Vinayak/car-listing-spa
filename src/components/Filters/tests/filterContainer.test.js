import React from "react";
import thunk from "redux-thunk";
import Filter from "../filterContainer";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";

const middlewares = [thunk];

const mockStore = configureMockStore(middlewares);

let store = mockStore({
  filters: {
    color: "red",
    manufacturer: "BMW"
  }
});

describe("Filter component", () => {
  it("shallow renders without crashing", () => {
    const tree = renderer.create(<Filter store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
