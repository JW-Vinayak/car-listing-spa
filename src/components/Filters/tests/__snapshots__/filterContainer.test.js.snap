// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component shallow renders without crashing 1`] = `
<div
  className="filters"
>
  <div
    className="filter"
  >
    <span
      className="filter-name"
    >
      Color
    </span>
    <select
      className="filter-dropdown"
      onChange={[Function]}
      value="red"
    >
      <option
        name="default"
        value=""
      >
        All car colors
      </option>
    </select>
  </div>
  <div
    className="filter"
  >
    <span
      className="filter-name"
    >
      Manufacturer
    </span>
    <select
      className="filter-dropdown"
      onChange={[Function]}
      value="BMW"
    >
      <option
        name="default"
        value=""
      >
        All manufacturers
      </option>
    </select>
  </div>
  <button
    className="apply-button button"
    onClick={[Function]}
  >
    Filter
  </button>
</div>
`;
